PopPUNK/visualise.py

```diff
@@ -310,19 +325,108 @@ def generate_visualisations(query_db,
                 return_dict = True)
         isolateClustering = joinClusterDicts(isolateClustering, queryIsolateClustering)
 
+    # Generate MST
+    mst_tree = None
+    if mst:
+        existing_tree = None
+        if not overwrite:
+            existing_tree = check_tree_exists(output, "MST")
+        if existing_tree is None:
+            G = None
+            if sparse_distMat is not None:
+                sys.stderr.write("Started sparse distance matrix processing\n")
+                combined_dists = scipy.sparse.load_npz(sparse_distMat)
+                sys.stderr.write("Completed sparse distance matrix processing\n")
+                sys.stderr.write("Started constructing partial graph\n")
+                G = constructNetwork(combined_seq,
+                                     combined_seq,
+                                     [0]*combined_dists.shape[0],
+                                     0,
+                                     edge_list=False,
+                                     sparse=True,
+                                     sparse_input=combined_dists)
+                
+                sys.stderr.write("Completed constructing partial graph\n")
+            else:
+                sys.stderr.write("Started distance matrix processing\n")
+                if qr_distMat is not None: # not sure how to concatenate QR/QQ matrices to sparse matrix
+                    combined_dists = np.concatenate((rr_distMat, qr_distMat, qq_distMat), axis = 0)
+                else:
+                    combined_dists = rr_distMat
```

Comment:
We should look at this, but probably we want to move the code from `LineageFit.extend()` into its own function, and call it from there and also here.

I think `np.concatenate` will make everything dense as a first step